### PR TITLE
13480 - Added conditional to receivingOrgSvcStatus

### LINKED ISSUE #13480: Add Receiver Status (active or inactive) to the API return call to ensure only active receivers will be shown on Daily Data
## Problem statement
Currently, the API call on the Daily Data page returns both Inactive and Active receivers when presented with no filter. The Daily Data page should only show active receivers, so changing the API call is needed.

## What you need to know

- Users do not need to see inactive receivers on their daily data page
- We can add the functionality to see all receivers (inactive and active) in the future

## Acceptance criteria

- [x] API updated to allow Daily Data page to load showing only active receivers in the table even without any receiver service selected.

### PR DESCRIPTION
**Pull request by penny-lischer**
This PR ...

Adds conditional logic for the deliveries receivingOrgSvcStatus param.

Test Steps:
1. Both of these API calls should return data
`/api/waters/org/ca-dph/deliveries?sortDir=DESC&sortColumn=batchReadyAt&since=2023-11-30T00:00:00.000Z&until=2023-12-31T00:00:00.000Z&receivingOrgSvcStatus=ACTIVE&receivingOrgSvcStatus=TESTING`

`/api/waters/org/ca-dph/deliveries?sortDir=DESC&sortColumn=batchReadyAt&since=2023-11-30T00:00:00.000Z&until=2023-12-31T00:00:00.000Z`

### Testing
- [x] Tested locally?
- [ ] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [ ] Added tests?

## Linked Issues
- Fixes #13480

### CODE REVIEW

**Review by victor-chaparro — APPROVED** [review]

File: prime-router/src/main/kotlin/history/azure/HistoryDatabaseAccess.kt [hunk 1]
```diff
@@ -229,11 +229,13 @@ abstract class HistoryDatabaseAccess(
     ): Condition {
         var filter = this.organizationFilter(organization, orgService)
 
-        var statusList = receivingOrgSvcStatus?.map { it.name.lowercase() }
-        filter = filter.and(
+        if (receivingOrgSvcStatus != null) {
```
Comment (arnejduranovic) [inline prime-router/src/main/kotlin/history/azure/HistoryDatabaseAccess.kt:232]: Are there tests that verify this new logic?

**Review by arnejduranovic — COMMENTED** [review]
Changes look good but need to make sure we have tests

File: prime-router/src/main/kotlin/history/azure/HistoryDatabaseAccess.kt (hunk 1 above)
Comment (penny-lischer) [reply prime-router/src/main/kotlin/history/azure/HistoryDatabaseAccess.kt:232]: There are existing tests where if you do not pass this, the response status is validated.

**Review by arnejduranovic — APPROVED** [review]
Will add database tests for HistoryDatabaseAccess in another ticket

**Pull request merged**